substrate/client/rpc-spec-v2/src/common/connections.rs

```diff
@@ -0,0 +1,228 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use jsonrpsee::ConnectionId;
+use parking_lot::Mutex;
+use std::{
+	collections::{HashMap, HashSet},
+	sync::Arc,
+};
+
+/// Limit the RPC functionality to a single connection.
+#[derive(Default, Clone)]
+pub struct RpcConnections {
+	/// The number of tokens that can be registered for each connection.
+	capacity: usize,
+	/// Map the connecton ID to a set of tokens.
+	data: Arc<Mutex<HashMap<ConnectionId, ConnectionData>>>,
+}
+
+#[derive(Default)]
+struct ConnectionData {
+	/// The total number of tokens.
+	///
+	/// # Note
+	///
+	/// Because a pending subscription sink does not expose the future subscription ID,
```

Comment:
I wonder if we can make it explicit that token == subscription ID?

It may be hard to understand this comment about the subscription sink otherwise

Reply:
Indeed the `token` name is a bit misleading. This is the subscription ID in this case, however it will be the transaction `OperatationId` for the tx API 🤔 

Maybe `identifier` sounds better? 🤔